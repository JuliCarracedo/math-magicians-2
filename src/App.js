import './App.css';
import React from 'react';
import { BrowserRouter as Router, Routes, Switch } from 'react-router-dom';
import Navbar from './components/Navbar';
import Calculator from './components/Calculator';
import Home from './components/Home';

function App() {
  return (
    <React.StrictMode>
      <Router>
        <Navbar />
        <Routes>
          <Route exact path="/" render={() => <Home />} />
          <Route path="/calc" render={() => <Calculator />} />
        </Routes>
      </Router>
    </React.StrictMode>
  );
}

export default App;
