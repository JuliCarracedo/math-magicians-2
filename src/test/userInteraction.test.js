import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import App from '../App';

describe('test pages', () => {
  it('testing home page', () => {
    render(<App />);
    const Home = screen.getByText('Home');
    fireEvent.click(Home);
    expect(screen.getByText('Welcome to our page')).toMatchSnapshot();
  });
  it('test calculator page', () => {
    render(<App />);
    const Calculator = screen.getByText('Calculator');
    fireEvent.click(Calculator);
    expect(screen.getByText('Let\'s do some math!')).toMatchSnapshot();
  });
  it('test qoute page', () => {
    render(<App />);
    const Qoute = screen.getByText('Quote');
    fireEvent.click(Qoute);
    expect(screen.getByText('"Mathematics is language that God used to write the world" : Galileo Galilei')).toMatchSnapshot();
  });
});
