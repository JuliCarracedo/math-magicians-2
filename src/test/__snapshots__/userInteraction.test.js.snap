// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test pages test calculator page 1`] = `
<h2>
  Let's do some math!
</h2>
`;

exports[`test pages test qoute page 1`] = `
<p>
  "Mathematics is language that God used to write the world" : Galileo Galilei
</p>
`;

exports[`test pages testing home page 1`] = `
<h2>
  Welcome to our page
</h2>
`;
